File: leetcode/0003.go
package leetcode

func lengthOfLongestSubstring(s string) int {
	var m [256]int
	ans := 0
	for i, j := 0, 0; j < len(s); j++ {
		if k := m[s[j]]; k != 0 && k > i {
			i = k
		}
		if ans < j-i+1 {
			ans = j - i + 1
		}
		m[s[j]] = j + 1
	}
	return ans
}

File: leetcode/0003.cpp
class Solution {
public:
    int m[256];
    int lengthOfLongestSubstring(string s) {
        int len = s.length();
        int ans = 0;
        for (int i = 0, j = 0; i < len; ++i) {
            int k = m[s[i]];
            if (k != 0) {
                j = max(j, k);
            }
            ans = max(ans, i - j + 1);
            m[s[i]] = i + 1;
        }
        return ans;
    }
};